docs/architecture/grpc-for-wcf-developers/self-hosted.md

```diff
@@ -1,11 +1,195 @@
+
+```console
+sudo systemctl daemon-reload
+sudo systemctl status myapp
+```
+
+If you've configured the service correctly, the following output will be shown:
+
+```text
+myapp.service - My gRPC Application
+ Loaded: loaded (/etc/systemd/system/myapp.service; disabled; vendor preset: enabled)
+ Active: inactive (dead)
+```
+
+Use the `start` command to start the service.
+
+```console
+sudo systemctl start myapp.service
+```
+
+> [!TIP]
+> The `.service` extension is optional when using `systemctl start`.
+
+To tell `systemd` to start the service automatically on system startup, use the `enable` command.
+
+```console
+sudo systemctl enable myapp
+```
+
+### Log to journald
+
+The Linux equivalent of the Windows Event Log is `journald`, a structured logging system service that is part of `systemd`. Log messages written to the standard output by a Linux daemon are automatically written to `journald`, so to configure logging levels, use the `Console` section of the logging configuration. The `UseSystemd` host builder method automatically configures the console output format to suit the journal.
+
+Because `journald` is the standard for Linux logs, there are a variety of tools that integrate with it, and you can easily route logs from `journald` to an external logging system. Working locally on the host, you can use the `journalctl` command to view logs from the command line.
+
+```console
+sudo journalctl -u myapp
+```
+
+> [!TIP]
+> If you have a GUI environment available on your host, a few graphical log viewers are available for Linux, such as *QJournalctl* and *gnome-logs*.
+
+To learn more about querying the systemd journal from the command line with `journalctl`, see [the man pages](https://manpages.debian.org/buster/systemd/journalctl.1).
+
+## HTTPS Certificates for self-hosted applications
+
+When running a gRPC application in production, you should use a proper TLS certificate from a trusted Certificate Authority (CA). This CA could be a public CA, or an internal one for your organization. On Windows hosts, the certificate should be loaded from a secure [Certificate Store](https://docs.microsoft.com/windows/win32/seccrypto/managing-certificates-with-certificate-stores) using the [X509Store class](https://docs.microsoft.com/dotnet/api/system.security.cryptography.x509certificates.x509store?view=netcore-3.0). On Linux hosts, the certificate may be created using one of the [X509Certificate2 constructors](https://docs.microsoft.com/dotnet/api/system.security.cryptography.x509certificates.x509certificate.-ctor?view=netcore-3.0), either from a file (for example a `.pfx` file protected by a strong password), or from binary data retrieved from a secure storage service such as [Azure Key Vault](https://azure.microsoft.com/services/key-vault/).
```

Comment:
"proper" is ambiguous.
...you should use a TLS certificate signed by a trusted....
Also, if running on Windows in Azure, could the certificate by provided from KeyVault instead?